web/modules/custom/dpl_opening_hours/src/Plugin/rest/resource/v1/OpeningHoursLegacyResource.php

```diff
@@ -0,0 +1,251 @@
+    }
+
+  }
+
+  /**
+   * Map a value object to an OpenAPI response.
+   *
+   * @throws \Exception
+   */
+  public function toLegacyResponse(OpeningHoursInstance $instance) : OpeningHoursLegacyResponse {
+    return (new OpeningHoursLegacyResponse())
+      ->setNid(intval($instance->branch->id()))
+      ->setCategoryTid(intval($instance->categoryTerm->id()))
+      ->setDate(new DateTime($instance->startTime->format('Y-m-d')))
+      ->setStartTime($instance->startTime->format("H:i"))
+      ->setEndTime($instance->endTime->format('H:i'))
+      ->setNotice(NULL);
+  }
+
+  /**
+   * Generate a schema for an opening hours legacy instance.
+   *
+   * This allows for reuse across implementing classes.
+   *
+   * @return mixed[]
+   *   OpenAPI schema for a single opening hours legacy instance.
+   */
+  protected function openingHoursLegacyInstanceSchema(): array {
+    return [
+      "type" => "object",
+      "properties" => [
+        "nid" => [
+          "type" => "integer",
+          "description" => "The node Id of the branch the opening hours instance belongs to.",
+        ],
+        "category_tid" => [
+          "type" => "integer",
+          "description" => "The (t)id of the opening hours category.",
+        ],
+        "date" => [
+          "type" => "string",
+          "format" => "date",
+          "description" => "The date which the opening hours applies to. In ISO 8601 format.",
+        ],
+        "start_time" => [
+          "type" => "string",
+          "example" => "09:00",
+          "description" => "When the opening hours start. In format HH:MM",
+        ],
+        "end_time" => [
+          "type" => "string",
+          "example" => "17:00",
+          "description" => "When the opening hours end. In format HH:MM",
+        ],
+        "notice" => [
+          "type" => "string",
+          "description" => "The id for the branch the instance belongs to",
+        ],
+      ],
+      "required" => [
+        "nid",
+        "category_tid",
+        "date",
+        "start_time",
+        "end_time",
+        "notice",
+      ],
+    ];
+  }
+
+  /**
+   * Generate the format to use by the serializer from the request.
+   */
+  protected function serializerFormat(Request $request): string {
+    $contentTypeFormat = $request->getContentTypeFormat();
+    if (!$contentTypeFormat) {
+      // Default to JSON format. Some code generators will not provide a default
+      // value even though it is provided in the spec.
+      $contentTypeFormat = $request->get('_format', 'json');
+    }
+    $mimeType = $request->getMimeType($contentTypeFormat);
+    if (!$mimeType) {
+      throw new \InvalidArgumentException("Unable to identify serializer format from content type form: $contentTypeFormat");
+    }
+    return $mimeType;
+  }
+
+  /**
+   * Invalidate the response cache.
+   */
+  protected function invalidateCache(): void {
+    $this->cacheTagsInvalidator->invalidateTags([self::CACHE_TAG_LIST]);
+  }
```

Comment:
Please remove this function.

1. It is not used anywhere within this class
2. It replicates what is already provided by the parent class